test/libsinsp/sys_call_test.cpp

```diff
@@ -0,0 +1,1235 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+Copyright (C) 2024 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include "sys_call_test.h"
+
+#include "event_capture.h"
+#include "subprocess.h"
+
+#include <libscap/scap-int.h>
+#include <libscap/scap_platform.h>
+
+#include <gtest/gtest.h>
+
+#include <errno.h>
+#include <event.h>
+#include <fcntl.h>
+#include <ifaddrs.h>
+#include <netinet/in.h>
+#include <poll.h>
+#include <signal.h>
+#include <termios.h>
+#include <unistd.h>
+
+#include <sys/ioctl.h>
+#include <sys/mman.h>
+#include <sys/quota.h>
+#include <sys/signalfd.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <sys/syscall.h>
+#include <sys/time.h>
+#include <sys/timerfd.h>
+#include <sys/types.h>
+
+#include <algorithm>
+#include <cassert>
+#include <list>
+#include <numeric>
+
+using namespace std;
+
+TEST_F(sys_call_test, stat)
```

Comment:
Uhm, all the tests in this file seem to do the exact thing as `drivers_test`.
They run a syscall or a combination of syscalls and then assert that the event is here and something is happening in userspace... why don't we extend the `class event_test` to support some inspections in sinsp instead of introducing all these executables/classes + callbacks, WDYT?

Reply:
Although some tests are similar in this first batch, the whole contribution process of the test suite we foresee can potentially cover use cases that diverge with the goals of the driver-level tests. It's hard to predict it at the moment, and refactor everything right now may be outside of the engineering bandwidth we currently have available. Plus, having this suite being its own standalone thing will allow us to converge it with the drivers one once we have a more high-level view than now. If this was a library code contribution, I'd agree with you of this being a strong blocker. But since this is a test-based contribution, I think it can provide tons of value as-is and we can postpone cleaning it up deeply as a secondary priority. Moreover @therealbobo is also curating each test while contributing them.